metadata-ingestion/src/datahub/emitter/mcp_builder.py

```diff
@@ -35,20 +37,38 @@ def guid(self) -> str:
         return str(md5_hash.hexdigest())
 
 
-@dataclasses.dataclass
 class PlatformKey(DatahubKey):
     platform: str
-    instance: Optional[str]
+    instance: Optional[str] = None
+    environment: Optional[str] = None
+
+    @root_validator(pre=True)
+    def check_instance_environment(cls, values: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Okay this clears things up. Never an env inside..

Reply:
This makes it a bit confusing for folks trying to understand how container keys are generated. I would naturally think the contract is that this object is turned into a dict then hashed, but really theres some logic inside.